src/plugins/auto/plugin.cpp

```diff
@@ -679,7 +670,9 @@ QueryNetworkResult MultiDeviceInferencePlugin::QueryNetwork(const CNNNetwork&
     queryconfig.apply_user_properties();
     auto fullproperty = queryconfig.get_full_properties();
     // this can be updated when plugin switch to 2.0 API
-    std::map<std::string, std::string> fullConfig =  ConvertToStringMap(fullproperty);;
+    std::map<std::string, std::string> fullConfig =  ConvertToStringMap(fullproperty);
+    if (!queryconfig.is_set_by_user(ov::cache_dir))
+        fullConfig.erase(ov::cache_dir.name());
```

Comment:
will cache_dir affect query result?

Reply:
with current logic we have:
- AUTO asks about device supported properties via `ICore::get_supported_property` with `fullConfig`, which contains AUTOs' default value for `ov::cache_dir`
- Current PR changes the logic of `ICore::get_supported_property`, so it reports both core-level and plugin-level support properties.
- Then AUTO passes this supported option to `compile_model` / `query_model` calls. In `compile_model` call `ICore` removes core-level properties such as auto-batch settings or `ov::cache_dir`, while in other methods like `query_model` it does not expect `ov::cache_dir` (because it's not applicable here), and does not remove it. So, `cache_dir` is then passed to HW plugin, which does not expect it and throws exception.


Proper approach since this PR:
- Virtual plugins should not make any assumptions about properties of other devices or core-level properties. E.g. we except AUTO will remove default values for auto-batch plugin settings or cache_dir at all.
- Virtual plugins will report `supported_properties` as:
    - Only their supported properties if `ov::device::priorities` is not passed
    - Their + underlying plugin supported properties if `ov::device::priorities` is passed. It will allow `AUTO` (or any other virtual plugin), just pass ALL the user settings to `ICore::get_supported_property` which will return a set of properties, that needs to be passed to next `get_core()->compile_model(...)` / `get_core()->query_model(...)` calls.

Reply:
Removed here https://github.com/openvinotoolkit/openvino/pull/16775